### ISSUE TITLE
saspy.SASdata.to_df Method Slow

### ISSUE THREAD

**Issue by kevinjfoley** [sassoftware/saspy#65]
I've tested the to_df method to move data between SAS and Python and it seems extremely slow.  

Below is from a notebook notebook which demonstrates the problem

---

```python
import saspy
import pandas as pd
import time 
import tempfile
```

```python
sas = saspy.SASsession(cfgname="winlocal")
sas.set_batch(True)
```

For example I created the following dataset with 1 million rows:

```python
generate_data = sas.submit('DATA test; DO i = 1 TO 1000000; j = i; k = j; l = k; m = "test"; OUTPUT; END; RUN;')
# print(generate_data["LOG"])
sas_dataset = sas.sasdata('test')
```

I then move the data to Python using the to_df method:

```python
start = time.time()
sas_df = sas_dataset.to_df()
end = time.time()
print(end - start)
```

    859.6239538192749
    

For comparison I wrote the following function to export the data to CSV and then import the CSV using pandas:

```python
def sas_to_df(sas_data_set):
    
    with tempfile.NamedTemporaryFile(delete=False) as temp_csv:
        temp_csv.close()
        sas_csv = sas_data_set.to_csv(temp_csv.name)
    
        df = pd.read_csv(temp_csv.name)
    
    return df
```

```python
start = time.time()
csv_df = sas_to_df(sas_dataset)
end = time.time()
print(end - start)
```

    2.186218738555908

**Comment by tomweber-sas** [comment]
Hey, does the code you're using have the fix from issue #39 ? I think it should be in the 2.1.7, although there a a couple other fixes with this that didn't make it into the pypi 2.1.7. They are commits at master. Can you pull from master and try that out? I'm hoping this is a dup of #39 :)

Thanks,
Tom

**Comment by kevinjfoley** [comment]
Thanks Tom, installing from master helps a lot, the time went down to around 9 seconds.
